doc/adr/0004-add-indexes-to-the-mongodb-image.md

```diff
@@ -0,0 +1,21 @@
+# 4. Add Indexes to the MongoDB image
+
+Date: 2017-03-21
+
+## Status
+
+Accepted
+
+## Context
+
+A spike has been done to check if the image can be layered, i.e. have the base image and then add the index as another layer, so
+the different applications would only use the image with the functionality they need.
+Unfortunately, due to the volatility of the data layer, another layer cannot modify it without making a copy, which is unsustainable.
+
+## Decision
+
+The base image had indexes added to it as even though not all the client applications needed them, the overhead was minimal.
+
+## Consequences
+
+Keeping a single source of truth for the data for both the client applications with the added search functionality.
```

Comment:
The consequences of doing `gp-finder` specific stuff in this repo are (possibly others):
* There might be contention between that app and other consumers
* The changes might be detrimental to other consumers and as they are updated independently this would be difficult to recognise